homeassistant/components/nest/climate_sdm.py

```diff
@@ -252,23 +246,28 @@ def preset_modes(self) -> list[str]:
     @property
     def fan_mode(self) -> str:
         """Return the current fan mode."""
-        if FanTrait.NAME in self._device.traits:
+        if (
+            self.supported_features & ClimateEntityFeature.FAN_MODE
+            and FanTrait.NAME in self._device.traits
+        ):
             trait = self._device.traits[FanTrait.NAME]
             return FAN_MODE_MAP.get(trait.timer_mode, FAN_OFF)
         return FAN_OFF
 
     @property
     def fan_modes(self) -> list[str]:
         """Return the list of available fan modes."""
-        modes = []
-        if FanTrait.NAME in self._device.traits:
-            modes = list(FAN_INV_MODE_MAP)
-        return modes
+        if (
+            self.supported_features & ClimateEntityFeature.FAN_MODE
+            and FanTrait.NAME in self._device.traits
+        ):
+            return list(FAN_INV_MODE_MAP)
```

Comment:
Side note: It's nice to create a constant for the list so we don't need to create new lists on every property access.

Reply:
Yes that's right. It's not going to work in different HVAC nodes.

Should let the user try and just throw home assistant errors instead?